### ISSUE TITLE
how to solve differential equation using chaospy tool in py35 environment ?

### ISSUE THREAD

**Issue by rajshekharM** [jonathf/chaospy#72]
for example how to solve the PV=nRT gas equation for error analysis of different experimental variable errors in p, V and T measurements

**Comment by jonathf** [comment]
From what little you are saying, it might sound like you are dealing with an inverse problem, not an forward one. If that is the case, I can not help you. Chaospy only addresses uncertainty in forward problems. But if you are talking about forward problems, continue reading.

There are more than one way. If you want the no-fuzz variant, take a look at the tutorial:
http://chaospy.readthedocs.io/en/master/tutorial.html

In your scenario you need to construct a function `foo`, that creates deterministic solutions to the gas equation. I haven't worked with gass equations, but I guess you can make something simple like:
```
import scipy.linalg
def foo(P, V, R, T):
   return scipy.linalg.solve(P*V, R*T)
```
Create random variables for the variables of interests, and numpy arrays (or constants) for the rests. It should be fairly easy to follow the tutorial given this. I recommend starting with collocation method. Fewer things that can go wrong with it.

**Comment by projectheartbeat** [comment]
Thanks for the answer !! Saved my life as I was stuck on this for a long time.
Just the one thing: As it seemed I was a bit vague, if I could more specific: In the attached book , in example 4.2, can we derive equation 4.11 from equation 4.10 using python and get the result of example 4.2 ? I tried with chaospy (but in vain), can we use any other tools and packages for the same (even chaospy can be ok, as long as it produces same result) ? 


[Experimentation, Validation, an - Hugh W. Coleman.pdf](https://github.com/jonathf/chaospy/files/1704429/Experimentation.Validation.an.-.Hugh.W.Coleman.pdf)

**Comment by jonathf** [comment]
That is some book you got there. I skimmed and I am unsure what this author was thinking. The problem here is one of language. The book is referring to the terms "uncertainty" and "sensitivity", but they are ambiguous terms if they are not defined properly. The author does not properly define the term, and instead jumps forth and back about what that means between chapters. There is no consistent rule between when what is used. I doubt this book had a proper critical review by anyone with a formal background in mathematics or statistics.

I will say that in chapter 2 and 3, though not properly defined, is mostly on target, though. This is probability theory, and I understand why you would deduce that Chaospy could be applicable for the problems described. However in chapter 4, uncertainty gets redefined and he start playing around with (gradient based) sensitivity analysis. (Or strictly speaking a simplified variant of SA where any model is replaced with first order Taylor series.) SA is not based on probabilities. They are not really related except for in name. Doesn't help if you take your coefficients and call them "relative uncertainties". (Sidenote: If a probabilistic approach to sensitivity is wanted, _variance based_ sensitivity is possible, but that is a different framework.)

In other words when the book starts talking about "relative uncertainty" in chapter 4., those are probabilistic, and you can neither use Chaospy nor Monte Carlo to address the problems described. It is simply a different topic.

**Comment by jonathf** [comment]
Sorry for the rant.

If you something is unclear or there is something else you want to address, feel free to reopen. Otherwise I consider this issue closed.
